### PR TITLE
Remove visible search option for legacy packages.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

**Review by jonasfj — APPROVED** [review]

File: doc/help-search.md [hunk 1]
```diff
@@ -36,3 +36,16 @@ The search UI also supports filters:
...
 specificity and the text match score of the search team.
+
+## Legacy packages
+
+`pub.dev` does not expose packages that support only Dart SDK 1.x in search results
+or in listing pages. One can access these packages through their package page
+(`/packages/<pkg>`) and `pub` client can download them.
+
+You can use the following options to search for these packages:
+- Add `legacy=1` query parameter to the URL ([example](https://pub.dev/packages?legacy=1)).
```
Comment (jonasfj) [inline doc/help-search.md:47]: I suggest we don't document and remove the query parameter... I know that just leaves `is:legacy`, but IMO that's fine..

Comment (isoos) [reply doc/help-search.md:47]: `is:legacy` doesn't work on itself - not sure why, need to investigate it.

Comment (jonasfj) [reply doc/help-search.md:47]: Isn't it better to just file a bug for that and fix it independently, I don't think it's super-bad if that is temporarily broken.

Comment (isoos) [reply doc/help-search.md]: Filed https://github.com/dart-lang/pub-dev/issues/4275

**Pull request merged**